cli/src/pcluster/templates/imagebuilder_stack.py

```diff
@@ -0,0 +1,186 @@
+# Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
+# with the License. A copy of the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+# OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions and
+# limitations under the License.
+
+#
+# This module contains all the classes required to convert a ImageBuilder into a CFN template by using CDK.
+#
+
+import os
+
+from aws_cdk import aws_iam as iam
+from aws_cdk import aws_imagebuilder as imagebuilder
+from aws_cdk import aws_ssm as ssm
+from aws_cdk import core
+
+import pcluster.utils as utils
+from common.utils import load_yaml
+from pcluster.models.imagebuilder import ImageBuilder
+
+
+class ImageBuilderStack(core.Stack):
+    """Create the Stack for imagebuilder."""
+
+    def __init__(self, scope: core.Construct, construct_id: str, imagebuild: ImageBuilder, **kwargs) -> None:
+        super().__init__(scope, construct_id, **kwargs)
+        image = imagebuild.image
+        build = imagebuild.build
+        dev_settings = imagebuild.dev_settings
+
+        # TODO: use attributes from imagebuilder config instead of using these static variables.
+        core.CfnParameter(self, "EnableNvidia", type="String", default="false", description="EnableNvidia")
+        core.CfnParameter(self, "EnableDCV", type="String", default="false", description="EnableDCV")
+        default_node_url = dev_settings.node_url if dev_settings and dev_settings.node_url else ""
+        core.CfnParameter(
+            self,
+            "CustomNodePackage",
+            type="String",
+            default=default_node_url,
+            description="CustomNodePackage",
+        )
+        core.CfnParameter(
+            self,
+            "UpdateAndReboot",
+            type="String",
+            default=str.lower(str(bool(dev_settings.update_os_and_reboot))),
+            description="UpdateAndReboot",
+        )
+        ami_info = utils.get_info_for_amis([build.parent_image])
+        architecture = ami_info[0].get("Architecture")
+        if build.instance_type:
+            instance_types = [build.instance_type]
+        else:
```

Comment:
The `build.instance_type` is a required param. There should be no default. Instead there should be a validator to check if the instance type used is compatible with the parent image